src/Ethereum/ABI/Array.h

```diff
@@ -46,4 +47,40 @@ class ParamArray: public ParamCollection
     virtual std::string getExtraTypes(std::vector<std::string>& ignoreList) const;
 };
 
+/// Fixed-size array of the same type e.g, "type[4]"
+class ParamArrayFix final : public ParamCollection {
+public:
+    //! Public constants
+    static constexpr std::size_t paddingSize{32};
+
+    //! Public Definitions
+    using Params = std::vector<std::shared_ptr<ParamBase>>;
+
+    //! Public constructor
+    explicit ParamArrayFix(std::size_t n, const Params& params) noexcept(false)
+        : ParamCollection(), _n(n) {
+        if (params.size() != n) {
+            throw std::runtime_error("params need to have the same size as N");
+        }
+        this->addParams(params);
+    }
+
+    //! Public member methods
+    [[nodiscard]] std::size_t getCount() const final { return _n; }
+    [[nodiscard]] size_t getSize() const final { return paddingSize + _params.getSize(); }
```

Comment:
Is padding always added? ParamSet.getSize() does not inlcude padding?

Reply:
It's seems so, I guess `params.getSize()` should be equal to `hex(encoded).size()` alright?

Reply:
Padding should be applied, but that does not mean 32 has to be always added to the size!
I think correct would be 
```
[[nodiscard]] size_t getSize() const final { return ValueEncoder::paddedTo32(_params.getSize()); }
```

Reply:
Actually this is not needed, ParamSet::getSize() always pads.
So simply:
```
[[nodiscard]] size_t getSize() const final { return _params.getSize(); }
```